src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/DataReaderTest/DataReaderStreamsTest.cs

```diff
@@ -0,0 +1,611 @@
+            Assert.Equal(originalText, outputText);
+        }
+
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
+        [ConditionalTheory(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringsSetup))]
+        [MemberData(nameof(GetCommandBehaviourAndAccessorTypes))]
+        public static void NullStreamProperties(CommandBehavior behavior, AccessorType accessorType)
+        {
+            string query = "SELECT convert(xml,NULL) AS XmlData, convert(nvarchar(max),NULL) as TextData, convert(varbinary(max),NULL) as StreamData";
+
+            using (SqlConnection connection = new SqlConnection(DataTestUtility.TCPConnectionString))
+            using (SqlCommand command = new SqlCommand(query, connection))
+            {
+                connection.Open();
+
+                // do clean queries to get field values again in case of sequential mode
+
+                using (SqlDataReader reader = command.ExecuteReader(behavior))
+                {
+                    if (reader.Read())
+                    {
+                        Assert.True(reader.IsDBNull(0));
+                        Assert.True(reader.IsDBNull(1));
+                        Assert.True(reader.IsDBNull(2));
+                    }
+                }
+
+                using (SqlDataReader reader = command.ExecuteReader(behavior))
+                {
+                    if (reader.Read())
+                    {
+                        Assert.True(reader.IsDBNullAsync(0).GetAwaiter().GetResult());
+                        Assert.True(reader.IsDBNullAsync(1).GetAwaiter().GetResult());
+                        Assert.True(reader.IsDBNullAsync(2).GetAwaiter().GetResult());
+                    }
+                }
+
+                using (SqlDataReader reader = command.ExecuteReader(behavior))
+                {
+                    if (reader.Read())
+                    {
+                        using (XmlReader xmlReader = GetValue<XmlReader>(reader, 0, accessorType))
+                        {
+                            Assert.NotNull(xmlReader);
+                            Assert.Equal(accessorType == AccessorType.GetFieldValueAsync, xmlReader.Settings.Async);
+                            Assert.Equal(xmlReader.Value, string.Empty);
+                            Assert.False(xmlReader.Read());
+                            Assert.True(xmlReader.EOF);
+                        }
+
+                        using (TextReader textReader = GetValue<TextReader>(reader, 1, accessorType))
+                        {
+                            Assert.NotNull(textReader);
+                            Assert.True(behavior != CommandBehavior.SequentialAccess || textReader.GetType().Name.Contains("Sequential"));
+                            Assert.Equal(textReader.ReadToEnd(), string.Empty);
+                        }
+
+                        using (Stream stream = GetValue<Stream>(reader, 2, accessorType))
+                        {
```

Comment:
Do we also want to check if the stream is empty here?

Reply:
Empty checks are handled in the follow block with a clean reader. You have to be careful of the read state of the streams which is why it's all split apart the way it is.